packages/toolkit/src/view/pipeline/view-pipeline/PipelineRunViews/List.tsx

```diff
@@ -0,0 +1,269 @@
+import * as React from "react";
+import Link from "next/link";
+
+import {
+  ColumnDef,
+  DataTable,
+  PaginationState,
+} from "@instill-ai/design-system";
+
+import {
+  EmptyView,
+  LoadingSpin,
+  RunsTableSortableColHeader,
+  RunStateLabel,
+} from "../../../../components";
+import {
+  convertToSecondsAndMilliseconds,
+  InstillStore,
+  useInstillStore,
+  useRouteInfo,
+  useShallow,
+} from "../../../../lib";
+import { usePaginatedPipelineRuns } from "../../../../lib/react-query-service/pipeline";
+import { env, getHumanReadableStringFromTime } from "../../../../server";
+import { TABLE_PAGE_SIZE } from "../constants";
+
+const selector = (store: InstillStore) => ({
+  accessToken: store.accessToken,
+  enabledQuery: store.enabledQuery,
+});
+
+export type PipelineRunListProps = {
+  pipeline?: Pipeline;
+};
+
+export const PipelineRunList = ({ pipeline }: PipelineRunListProps) => {
+  const [orderBy, setOrderBy] = React.useState<string>();
+  const { accessToken, enabledQuery } = useInstillStore(useShallow(selector));
+  const routeInfo = useRouteInfo();
+  const [paginationState, setPaginationState] = React.useState<PaginationState>(
+    {
+      pageIndex: 0,
+      pageSize: TABLE_PAGE_SIZE,
+    },
+  );
+  const pipelineRuns = usePaginatedPipelineRuns({
+    pipelineName: `namespaces/${routeInfo.data.namespaceId}/pipelines/${routeInfo.data.resourceId}`,
+    enabled: enabledQuery && routeInfo.isSuccess,
+    accessToken,
+    pageSize: TABLE_PAGE_SIZE,
+    page: paginationState.pageIndex,
+    orderBy,
+  });
+
+  const ownerId = React.useMemo(() => {
+    return pipeline?.name.split("/")[1];
+  }, [pipeline]);
+
+  const pageCount = React.useMemo(() => {
+    if (!pipelineRuns.data) {
+      return 1;
+    }
+
+    return Math.ceil(pipelineRuns.data.totalSize / pipelineRuns.data.pageSize);
+  }, [pipelineRuns.isSuccess, pipelineRuns.data]);
+
+  const onSortOrderUpdate = (sortValue: string) => {
+    setPaginationState((currentValue) => ({
+      ...currentValue,
+      pageIndex: 0,
+    }));
+    setOrderBy(sortValue);
+  };
+
+  const columns = React.useMemo(() => {
+    const columns: ColumnDef<PipelineRun>[] = [
+      {
+        accessorKey: "pipelineRunUid",
+        header: () => <div className="text-left">Run ID</div>,
+        cell: ({ row }) => {
+          return (
+            <div className="font-normal text-semantic-bg-secondary-secondary break-all">
+              <Link
+                href={`/${ownerId}/pipelines/${pipeline?.id}/runs/${row.getValue("pipelineRunUid")}`}
+                className="text-semantic-accent-default hover:underline"
+              >
+                {row.getValue("pipelineRunUid")}
+              </Link>
+            </div>
+          );
+        },
+      },
+      {
+        accessorKey: "pipelineVersion",
+        header: () => <div className="text-left">Version</div>,
+        cell: ({ row }) => {
+          let content: React.ReactNode = `Unversioned`;
```

Comment:
@orangecoloured I think you don't need to case this type here

Reply:
<img width="667" alt="Screenshot 2024-08-26 at 11 51 41 AM" src="https://github.com/user-attachments/assets/be72a636-50e2-41dc-8427-c750c6827d8f">

Reply:
Oh ok, got your point